py-polars/polars/io.py

```diff
@@ -705,6 +705,68 @@ def scan_parquet(
     )
 
 
+def scan_json(
+    file: str | Path,
+    infer_schema_length: int | None = 100,
+    batch_size: int | None = 1024,
+    n_rows: int | None = None,
+    skip_rows: int | None = None,
+    low_memory: bool = False,
+    rechunk: bool = True,
+    row_count_name: str | None = None,
+    row_count_offset: int = 0,
+    **kwargs: Any,
+) -> LazyFrame:
+    """
+    Lazily read from a JSON file.
+
+
+    This allows the query optimizer to push down predicates and projections to the scan
+    level, thereby potentially reducing memory overhead.
+    Note: `scan_json` currently only supports newline-delimited JSON files.
+
+    Parameters
+    ----------
+    file
+        Path to a file.
+    infer_schema_length
+        Infer the schema length from the first ``infer_schema_length`` rows.
+    batch_size
+        Number of rows to read in each batch.
+    n_rows
+        Stop reading from JSON file after reading ``n_rows``.
+    skip_rows
+        Skip the first ``skip_rows`` rows.
+    low_memory
+        Reduce memory pressure at the expense of performance.
+    rechunk
+        Reallocate to contiguous memory when all chunks/ files are parsed.
+    row_count_name
+        If not None, this will insert a row count column with give name into the
+        DataFrame
+    row_count_offset
+        Offset to start the row_count column (only use if the name is set)
+
+    """
+    # Map legacy arguments to current ones and remove them from kwargs.
+    n_rows = kwargs.pop("stop_after_n_rows", n_rows)
```

Comment:
If we are introducing new functionality, we don't need to provide support for legacy arguments. In fact, we should remove the `**kwargs` parameter as it is only used to extract `stop_after_n_rows`